File: 0932-monotonic-array/0932-monotonic-array.java
class Solution {
    boolean inc(int[]nums){
        for(int i=0;i<nums.length-1;i++)
        {
            if(nums[i+1]>nums[i])
                return false;
        }
          return true;   
    }
    boolean dec(int[] nums){
        for(int i=0;i<nums.length-1;i++)
        {
            if(nums[i]>nums[i+1])
                return false;
        }
          return true;
    }
    public boolean isMonotonic(int[] nums) {
        return (inc(nums) || dec(nums)) ;   
    }
}

File: 0932-monotonic-array/0932-monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        int n=nums.size();
        bool a=false,b=false;
        for(int i=0;i<n-1;i++)
        {
            if(nums[i]>nums[i+1])
            a=true;
            else if(nums[i+1]>nums[i])
            b=true;
            if(a==true & b==true)
            return false;
        }
        return true;
    }
};